### PR TITLE
CE-539: fix ammo not showing up when losing weapons in natla's mines

### PR DESCRIPTION
**Pull request by serishema**
CE-539

### CODE REVIEW

File: src/menu/compassmenustate.cpp [hunk 1]
```diff
@@ -126,7 +126,7 @@ CompassMenuState::CompassMenuState(const std::shared_ptr<MenuRingTransform>& rin
   auto addAmmoStats = [this, &inv](const char* weaponName, engine::WeaponType weaponType, size_t& y0)
   {
     const auto& ammo = inv.getAmmo(weaponType);
-    if(inv.count(ammo.weaponType) == 0 && ammo.ammo && ammo.hits == 0 && ammo.hitsTotal == 0 && ammo.misses == 0
+    if(inv.count(ammo.weaponType) == 0 && ammo.shots && ammo.hits == 0 && ammo.hitsTotal == 0 && ammo.misses == 0
```
Comment (stohrendorf) [inline src/menu/compassmenustate.cpp:129]:
```suggestion
    if(inv.count(ammo.weaponType) == 0 && ammo.shots != 0 && ammo.hits == 0 && ammo.hitsTotal == 0 && ammo.misses == 0
```

Forgot to write an explicit comparison there when I wrote this.

**Review by stohrendorf — CHANGES_REQUESTED** [review]
Didn't test this yet, but overall looks very good.

File: src/engine/objects/laraobject.cpp [hunk 2]
```diff
@@ -1234,23 +1232,34 @@ void LaraObject::updateAnimTwoWeapons(const WeaponType weaponType)
...
     return false;
   }
 
-  --ammo.ammo;
+  --ammo.shots;
+  shootBullet(weaponType, targetObject, weaponHolder, aimAngle);
+  return true;
+}
+
+void LaraObject::shootBullet(const WeaponType weaponType,
```
[the remark this answers is not in the document]
Comment (serishema) [reply src/engine/objects/laraobject.cpp:1254]: This algorithm is commonly called "hitscan" by gamers. 
so `hitscanSingleRound` comes to mind.

[the remark this answers is not in the document]
Comment (stohrendorf) [reply src/engine/objects/laraobject.cpp]: That's perfect.

File: src/engine/inventory.cpp [hunk 3]
```diff
@@ -24,22 +24,11 @@ size_t Inventory::put(const core::TypeId& id, world::World* world, const size_t
 {
   BOOST_LOG_TRIVIAL(debug) << "Object " << toString(id.get_as<TR1ItemId>()) << " added to inventory";
 
-  auto addWeapon = [this](Ammo& ammo, size_t n)
+  auto addWeapon = [this](Ammo& ammo)
```
Comment (stohrendorf) [inline src/engine/inventory.cpp:27]:
```suggestion
  auto addWeapon = [this](const Ammo& ammo)
```

**Review by stohrendorf — CHANGES_REQUESTED** [review]
Just one last thing.

**Review by stohrendorf — APPROVED** [review]
Good. Didn't test it yet, but if you're confident it works, you can squash, rebase and merge.

**Pull request merged**